salt/modules/swarm.py

```diff
@@ -0,0 +1,440 @@
+        return salt_return
+    except TypeError:
+        salt_return = {}
+        salt_return.update({'Error': 'Please make sure this minion is not part of a swarm and your passing remote_addr, listen_addr and token correctly.'})
+        return salt_return
+
+
+def leave_swarm(force=bool):
+    '''
+    Force the minion to leave the swarm
+
+    force
+      
+      Will force the minion/worker/manager to leave the swarm
+
+    CLI Example:
+
+    .. code-block:: bash
+      
+      salt '*' swarm.leave_swarm force=False
+    '''
+    salt_return = {}
+    client.swarm.leave(force=force)
+    output = server_name + ' has left the swarm'
+    salt_return.update({'Comment': output})
+    return salt_return
+
+
+def service_create(image=str,
+                   name=str,
+                   command=str,
+                   hostname=str,
+                   replicas=int,
+                   target_port=int,
+                   published_port=int):
+    '''
+    Create Docker Swarm Service Create
+
+    image
+    
+      the docker image
+
+    name
+      
+      is the service name
+
+    command
+     
+      the docker command to run in the container at launch
+
+    hostname 
+      
+      the hostname of the containers
+
+    replicas 
+     
+       how many replicas you want running in the swarm
+
+    target_port 
+    
+      the target port on the container
+
+    published_port  
+    
+      port thats published on the host/os
+    
+    CLI Example:
+
+    .. code-block:: bash
+      salt '*' swarm.service_create image=httpd name=Test_Service command=None hostname=salthttpd replicas=6 target_port=80 published_port=80
+    '''
+    try:
+        salt_return = {}
+        replica_mode = docker.types.ServiceMode('replicated', replicas=replicas)
+        ports = docker.types.EndpointSpec(ports={target_port: published_port})
+        client.services.create(name=name,
+                               image=image,
+                               command=command,
+                               mode=replica_mode,
+                               endpoint_spec=ports)
+        echoback = server_name + ' has a Docker Swarm Service running named ' + name
+        salt_return.update({'Info': echoback,
+                            'Minion': server_name,
+                            'Name': name,
+                            'Image': image,
+                            'Command': command,
+                            'Hostname': hostname,
+                            'Replicas': replicas,
+                            'Target_Port': target_port,
+                            'Published_Port': published_port})
+        return salt_return
+    except TypeError:
+        salt_return = {}
+        salt_return.update({'Error': 'Please make sure your passing arguments correctly [image, name, command, hostname, replicas, target_port and published_port]'})
+        return salt_return
```

Comment:
This should probably be backed up one indent instead of having the same return statement in both the try and except block

Reply:
So more like this with also removing the return in the try block? I think im doing this through out the module.  

```
except TypeError:
        salt_return = {}
        salt_return.update({'Error': 'Please make sure your passing arguments correctly [image, name, command, hostname, replicas, target_port and published_port]'})
return salt_return

```

Reply:
yeah, just like that!

I would change it through the module, since it is just an extra line that doesn't do anything special.

Reply:
@gtmanfred I can fix the returns throughout the module, my second question with the pylint errors.  I have the `__init__():` function set, It seems that I need to call the `__context__` in that function. How would you go about that? Seem that pylint is saying there are variables that are undefined. So I must not be calling them right.

Reply:
`__context__` is just a dictionary that is loaded into all our modules.  and can be used throughout the module.

so just replace all instances of those two variables to `__context__['server_name']` and `__context__['client']`